From 33923962e8c9e36c79221d2fe19f1c69250b0541 Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Wed, 4 Jun 2014 19:12:27 -0400
Subject: [powercap] Fix build error with option -Werror=format-security

Message-id: <20140604191227.42955.95035.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 82560
O-Subject: [PATCH RHEL7.1 BZ1085575 5/8] PowerCap: Fix build error with option -Werror=format-security
Bugzilla: 1085575
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1085575

Description:
------------
Fix compile error with gcc option: -Werror=format-security for
dev_set_name(&control_type->dev, name).

Changed to dev_set_name(&control_type->dev, "%s", name).

Signed-off-by: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Acked-by: Kees Cook <keescook@chromium.org>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

RHEL Version Found:
-------------------
7.0

Upstream Status:
----------------
http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=08ff4cbe10da402e7c493ac72e298975f62a9e78

----------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/powercap/powercap_sys.c b/drivers/powercap/powercap_sys.c
index c22fa4c..90ec0e2 100644
--- a/drivers/powercap/powercap_sys.c
+++ b/drivers/powercap/powercap_sys.c
@@ -623,7 +623,7 @@ struct powercap_control_type *powercap_register_control_type(
  control_type->ops = ops;
  INIT_LIST_HEAD(&control_type->node);
  control_type->dev.class = &powercap_class;
- dev_set_name(&control_type->dev, name);
+ dev_set_name(&control_type->dev, "%s", name);
  result = device_register(&control_type->dev);
  if (result) {
   if (control_type->allocated)
-- 
1.7.1